### DISCUSSION TITLE
tquic C++ 如何使用多路径传输提高传输效率?

### DISCUSSION THREAD

**Discussion by lion117** [Tencent/tquic#307] [Q&A (问题和回答)]
# 背景
1. Android13
2. libtquic.a

# 需求
系统使用wifi和5G网络同时传输一个大文件.  数据优先走wifi链路. 当wifi网络受到干扰时(丢包, 延迟), 能够动态的将部分数据走5G传输. 希望在双链路传输时能够实时监控5G和wifi模块的流量状态等信息. 

# 问题
是否有相关的C++ demo可参考学习一下?

**Comment by glime-ai** [comment]
TQUIC 支持多路径传输，可以在一个连接中同时使用多个路径来提高性能和可靠性。根据您的需求，您可以使用 TQUIC 的多路径功能来实现 WiFi 和 5G 网络的动态切换和流量监控。

目前，TQUIC 提供了 Rust、C 和 C++ 的接口，您可以参考以下文档和示例来实现您的需求：

1. **多路径传输**：TQUIC 支持多路径传输，您可以在一个连接中同时使用多个路径。详细信息请参考 [多路径传输文档](https://datatracker.ietf.org/doc/html/draft-ietf-quic-multipath)。

2. **C++ 接口**：TQUIC 提供了 C++ 接口，您可以参考 [API 文档](https://tquic.net/zh/docs/category/api-reference) 来了解如何使用这些接口。

3. **性能测试**：您可以参考 [性能测试文档](https://tquic.net/zh/docs/further_readings/benchmark) 来了解如何进行性能测试和监控。

4. **示例代码**：您可以在 TQUIC 的 GitHub 仓库中找到一些示例代码，具体请参考 [TQUIC GitHub](https://github.com/tencent/tquic)。

如果您有具体的实现问题，建议您查看 TQUIC 的 [常见问题](https://tquic.net/zh/docs/category/frequently-asked-questions) 或在 GitHub 上提交 issue 以获取更多帮助。

This comment was generated by [Glime](https://glimelab.ai/?utm_source=44520fce49ba43798485a628d6928e22&utm_medium=github).

**Comment by iyangsj** [comment]
Rust语言示例程序支持多路径：https://github.com/Tencent/tquic/tree/develop/tools/src/bin
示例程序使用方式参见：https://tquic.net/docs/getting_started/demo

C++语言示例程序目前较简单，不支持多路径。后续版本会提供。

**Reply by lion117** [reply]
想了解一下目前主流使用TQUIC的方案是直接集成RUST库到C++程序里面? 
还是直接使用TQUIC 转义生成的C库进行集成?
